src/com/projectkorra/projectkorra/board/BendingBoardInstance.java

```diff
@@ -43,88 +101,131 @@ public BendingBoardInstance(final BendingPlayer bPlayer) {
 		bendingSlots = bendingBoard.registerNewObjective("Board Slots", "dummy", title);
 		bendingSlots.setDisplaySlot(DisplaySlot.SIDEBAR);
 		player.setScoreboard(bendingBoard);
+		
+		for (int i = 0; i < 9; ++i) {
+			slots[i] = new BoardSlot(bendingBoard, bendingSlots, i);
+		}
+		
+		prefix = ChatColor.stripColor(ConfigManager.languageConfig.get().getString("Board.Prefix.Text"));
+		emptySlot = ChatColor.translateAlternateColorCodes('&', ConfigManager.languageConfig.get().getString("Board.EmptySlot"));
+		miscSeparator = ChatColor.translateAlternateColorCodes('&', ConfigManager.languageConfig.get().getString("Board.MiscSeparator"));
 
-		Arrays.fill(cachedSlots, "");
+		updateColors();
 		updateAll();
 	}
+	
+	private ChatColor getElementColor() {
+		if (bendingPlayer.getElements().size() > 1) {
+			return Element.AVATAR.getColor().asBungee();
+		} else if (bendingPlayer.getElements().size() == 1) {
+			return bendingPlayer.getElements().get(0).getColor().asBungee();
+		} else {
+			return ChatColor.WHITE;
+		}
+	}
+	
+	private ChatColor getColor(String from, ChatColor def) {
+		if (from.equalsIgnoreCase("element")) {
+			return getElementColor();
+		}
+		
+		try {
+			return ChatColor.of(from);
+		} catch (Exception e) {
+			ProjectKorra.plugin.getLogger().warning("Couldn't parse board color from '" + from + "', using default!");
+			return def;
+		}
+	}
 
 	public void disableScoreboard() {
 		bendingBoard.clearSlot(DisplaySlot.SIDEBAR);
 		bendingSlots.unregister();
 		player.setScoreboard(Bukkit.getScoreboardManager().getMainScoreboard());
 	}
 
-	private void setSlot(int slot, String name, boolean cooldown) {
-		if (slot < 1 || slot > 9 || !player.getScoreboard().equals(bendingBoard)) return;
-		String prefix = ChatColor.translateAlternateColorCodes('&', ConfigManager.languageConfig.get().getString("Board.SelectionPrefix"));
-		StringBuilder sb = new StringBuilder(slot == selectedSlot ? prefix : String.join("", Collections.nCopies(ChatColor.stripColor(prefix).length(), " ")));
-		if (name == null || name.isEmpty()) {
-			String emptySlot = ChatColor.translateAlternateColorCodes('&', ConfigManager.languageConfig.get().getString("Board.EmptySlot").replaceAll("\\{slot_number\\}", "" + slot));
-			sb.append(emptySlot);
+	public void setSlot(int slot, String ability, boolean cooldown) {
```

Comment:
We should probably have a wrapper method for this with the signature `public void setSlot(int slot, String ability)` which passes in a default value of `false` for the cooldown parameter to this method.